app/src/main/java/org/schabi/newpipe/util/DeviceUtils.java

```diff
@@ -57,8 +58,11 @@ public static boolean isTv(final Context context) {
         final PackageManager pm = App.getApp().getPackageManager();
 
         // from doc: https://developer.android.com/training/tv/start/hardware.html#runtime-check
-        boolean isTv = ContextCompat.getSystemService(context, UiModeManager.class)
-                .getCurrentModeType() == Configuration.UI_MODE_TYPE_TELEVISION
+        @Nullable
+        UiModeManager uiModeManager =
+                (UiModeManager) context.getApplicationContext().getSystemService(UI_MODE_SERVICE);
```

Comment:
Any reason to prefer `UI_MODE_SERVICE`. Please provide a reason or a doc reference to cross verify this. Also, provide a test case that returns `null` for existing function as pointed in another comment.